torch/distributed/distributed_c10d.py

```diff
@@ -285,6 +291,7 @@ class Backend(str):
         XCCL: ProcessGroup.BackendType.XCCL,
         UCC: ProcessGroup.BackendType.UCC,
         MPI: ProcessGroup.BackendType.MPI,
+        HCCL: ProcessGroup.BackendType.CUSTOM,
```

Comment:
@H-Huang do you happen to know if we support having multiple vendor backends listed here pointing to Custom, or is it going to cause a conflict next time someone adds one?

Reply:
On second look, it doesn't look like we really use `backend_type_map` anymore. I think it was originally used so we could map the backend to a certain device to be used for `barrier` and `object collectives` but it looks like `_get_object_coll_device` handles that for us. So we should remove this data structure.

Reply:
issue to track: https://github.com/pytorch/pytorch/issues/147044